apps/studio/components/interfaces/DataWarehouse/CreateWarehouseCollection.tsx

```diff
@@ -0,0 +1,98 @@
+import { useParams } from 'common'
+import { useCreateCollection } from 'data/analytics/warehouse-collections-create-mutation'
+import { useRouter } from 'next/router'
+import React from 'react'
+import toast from 'react-hot-toast'
+import { Button, IconPlus, Input, Modal } from 'ui'
+
+type Props = {}
+
+export const CreateWarehouseCollectionModal = (props: Props) => {
+  const [isOpen, setIsOpen] = React.useState(false)
+  const router = useRouter()
+  const { ref } = useParams()
+
+  const {
+    mutateAsync: createCollection,
+    isLoading,
+    data: newCollection,
+  } = useCreateCollection({
+    projectRef: ref || 'default',
+    onSuccess: (data) => {
+      router.push(`/project/${ref}/logs/collections/${data.data?.token}`)
+    },
+  })
+
+  return (
+    <>
+      <Button
+        type="default"
+        className="justify-start flex-grow w-full"
+        icon={<IconPlus size="tiny" />}
+        onClick={() => {
+          setIsOpen(!isOpen)
+        }}
+      >
+        New collection
+      </Button>
+      <Modal
+        size="medium"
+        onCancel={() => setIsOpen(!isOpen)}
+        header="Create an event collection"
+        visible={isOpen}
+        hideFooter
+      >
+        <form
+          onSubmit={async (e) => {
+            e.preventDefault()
+            try {
+              const formData = new FormData(e.target as HTMLFormElement)
+              const values = {
+                name: formData.get('name') as string,
+              }
+              await createCollection(values)
+              toast.success(`Collection ${values.name} created`)
+            } catch (error) {
+              console.error(error)
+              toast.error(`Failed to create collection. Check the console for more details.`)
```

Comment:
an aside: is this really the best ux? If we're asking users to check the console, might as well display the error message in a modal or textbox or something. I'd opt for a simple stringify of the error and chuck it in an alert wrapped with a `<pre>` so that people can copy-paste the error out to support.

Reply:
this error handling shouldn't be here - it should be in createCollection itself, we won't need the try catch as well if we use mutate instead of mutateAsync

the toast.success should be in the onSuccess callback of the RQ mutation, and the error handling should be in the onError callback of the RQ mutation as well (all RQ mutations would have a default fallback error handler as well, so unless this component needs a specific error handling behaviour, we can leave it)

but yes - always surface the error message, don't let the user think 💪